searcher/searcher.go

```diff
@@ -407,16 +407,18 @@ func newSearcher(
 		return nil, err
 	}
 
-	opt := &index.IndexOptions{
-		ExcludeDotFiles: repo.ExcludeDotFiles,
-		SpecialFiles:    wd.SpecialFiles(),
-	}
 
 	rev, err := wd.PullOrClone(vcsDir, repo.Url)
 	if err != nil {
 		return nil, err
 	}
 
+	opt := &index.IndexOptions{
+		ExcludeDotFiles:           repo.ExcludeDotFiles,
+		SpecialFiles:              wd.SpecialFiles(),
+		AutoGeneratedFilePatterns: wd.AutoGeneratedFilePatterns(vcsDir),
+	}
```

Comment:
Just to clarify, you swapped the order of this block so that we don't create `opt` if there's an error pulling?

Reply:
ah i swapped it because i needed `wd.PullOrClone` to be executed first so that when `wd.AutoGeneratedFilePatterns(vcsDir)` is called, it can find the `.gitattributes` file. `wd.PullOrClone` will populate the directory so without it, the first run of this flow will not see accurate results for autogenerated files

Reply:
ahh gotcha, that makes sense.